workspace-manager-clienttests/src/main/java/scripts/testscripts/CreateGetDeleteControlledGcsBucket.java

```diff
@@ -136,15 +149,96 @@ public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspace
     logger.info("Retrieving bucket resource id {}", resourceId.toString());
     GcsBucketAttributes gotBucket = resourceApi.getBucket(getWorkspaceId(), resourceId);
     assertThat(gotBucket.getBucketName(), equalTo(bucket.getGcpBucket().getBucketName()));
+    assertThat(gotBucket.getBucketName(), equalTo(bucketName));
 
-    // TODO: Check access:
-    // - writer can add the file
-    // - writer can read the file
-    // - reader can read the file
-    // - reader cannot write a file
-    // - reader cannot delete the bucket
+    Storage ownerStorageClient = ClientTestUtils.getGcpStorageClient(testUser, projectId);
 
-    // Delete bucket
+    // Owner can write object to bucket
+    BlobId blobId = BlobId.of(bucketName, GCS_BLOB_NAME);
+    BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/plain").build();
+    Blob createdFile = ownerStorageClient.create(blobInfo, GCS_BLOB_CONTENT.getBytes());
+    logger.info("Wrote blob {} to bucket", createdFile.getBlobId());
+
+    // Owner can read the object they created
+    Blob retrievedFile = ownerStorageClient.get(blobId);
+    assertThat(retrievedFile.getBlobId(), equalTo(createdFile.getBlobId()));
+    logger.info("Read existing blob {} from bucket as owner", retrievedFile.getBlobId());
+
+    // Second user has not yet been added to the workspace, so calls will be rejected.
+    Storage readerStorageClient = ClientTestUtils.getGcpStorageClient(reader, projectId);
+
+    // Second user cannot read the object yet.
+    try {
+      readerStorageClient.get(blobId);
+      throw new IllegalStateException(
+          "User accessed a controlled workspace bucket without being a workspace member");
+    } catch (StorageException storageException) {
+      assertThat(storageException.getCode(), equalTo(HttpStatusCodes.STATUS_CODE_FORBIDDEN));
+      logger.info("User {} outside of workspace could not access bucket as expected", reader.userEmail);
```

Comment:
can we use `assertThrows` instead? Ditto other try catch assert blocks

Reply:
It looks like we previously didn't have JUnit assertions. I've added them to the WSM TestRunner project here, which gives more useful assertions